vm/src/descriptor.rs

```diff
@@ -0,0 +1,12 @@
+use crate::function::OptionalArg;
+use crate::pyobject::{PyObjectRef, PyRef, PyResult, PyValue};
+use crate::VirtualMachine;
+
+pub trait PyBuiltinDescriptor: PyValue {
+    fn get(
+        zelf: PyRef<Self>,
+        obj: PyObjectRef,
+        cls: OptionalArg<PyObjectRef>,
+        vm: &VirtualMachine,
+    ) -> PyResult;
```

Comment:
I was thinking automatically adding slot `descr_get` and method `__get__` in pyimpl if the class has impl trait, but struggling with it for now.

Reply:
I like that idea; I've got a ~7 month old branch that allows putting pyimpl on a trait; maybe we could finally integrate that somehow. Maybe just put a TODO so that it's clear what we're aiming for eventually?

Reply:
That sounds good. I think I am not good at derives code :) will you work on it again?